admin/pageaudit/report/models.py

```diff
@@ -93,26 +112,95 @@ class Url(models.Model):
 
     url = models.URLField(unique=True)
     lighthouse_run = models.ForeignKey('LighthouseRun',
-                                        related_name='url_lighthouse_run',
-                                        on_delete=models.SET_NULL,
-                                        null=True, blank=True)
+                                       related_name='url_lighthouse_run',
+                                       on_delete=models.SET_NULL,
+                                       null=True,
+                                       blank=True)
     url_kpi_average = models.ForeignKey('UrlKpiAverage',
                                         related_name='url_url_kpi_average',
                                         on_delete=models.SET_NULL,
-                                        null=True, blank=True)
+                                        null=True,
+                                        blank=True)
     inactive = models.BooleanField(default=False)
     sequence = models.PositiveIntegerField(default=0)
-    
+    owner = models.ForeignKey(UrlOwner,
+                              related_name='url_owner_url',
+                              on_delete=models.SET_NULL,
+                              null=True,
+                              blank=True)
+
+    # based on https://developer.mozilla.org/en-US/docs/Web/API/Location
+    protocol = models.CharField(max_length=8, blank=True)
+    host = models.CharField(max_length=255, blank=True)
+    hostname = models.CharField(max_length=255, blank=True)
+    port = models.IntegerField(default=None, null=True, blank=True)
+    pathname = models.CharField(max_length=255, null=True, blank=True)
+    search = models.CharField(max_length=255, null=True, blank=True)
+    hash = models.CharField(max_length=255, null=True, blank=True)
+    # Note: not supporting username password as we don't want to persist
+    # that data in the database
+    origin = models.CharField(max_length=255, blank=True)
+    parsed_url = models.URLField(blank=True)
+
+    url_paths = models.ManyToManyField('UrlPath', blank=True)
```

Comment:
we also store all of the extracted `url paths` and `search key/vals` in related tables. The django admin never saves these automatically. You'll need to select the new ones and tie them in, but once we build our own UI for this it won't be a manual process

Reply:
That is correct. it is annoying. See: https://timonweb.com/posts/many-to-many-field-save-method-and-the-django-admin/

Reply:
It took me a while to understand this, but I now have figured it out by doing (i.e. highlighting the appropriate segments for a URL, observing the filter results come through, and then de-highlighting and seeing that they do not come through).

For the follow-up UI, do you think it would be something like this? I picture that you can choose a batch of URLs or something to which to bind the segments. There is an automated process that attempts to determine the relationship between URL and segment for each one. Then it may present the results to you for review, and you can correct manually if problem is observed?

Reply:
I observe that if there are multiple URLs that have a path in common (e.g., for a group of URLs, path 0 is `us-en`), then there is a new path created for each URL, even though they are ostensibly identical. Do you think there would be some automated means by which the model instances can be pruned to ensure no duplicated pairing of sequence number and path name?